cmp/options.go

```diff
@@ -338,17 +338,19 @@ func (cm comparer) String() string {
 }
 
 // AllowUnexported returns an Option that forcibly allows operations on
-// unexported fields in certain structs, which are specified by passing in a
-// value of each struct type.
+// unexported fields in certain structs. Struct types with permitted visibility
+// are specified by passing in a value of the struct type or by passing in a
+// PackagePrefix value, which represents all struct types defined in a package
+// for which the prefix is match of.
 //
-// Users of this option must understand that comparing on unexported fields
-// from external packages is not safe since changes in the internal
-// implementation of some external package may cause the result of Equal
-// to unexpectedly change. However, it may be valid to use this option on types
-// defined in an internal package where the semantic meaning of an unexported
-// field is in the control of the user.
+// Users must understand that comparing unexported fields from external packages
+// that are not owned by the user is unsafe since changes in the internal
+// implementation of external packages may cause the result of Equal to
+// unexpectedly change. However, it may be valid to use this option on types
+// owned by the user where the semantic meaning of an unexported field is in
+// full control of the user.
```

Comment:
Proposed simplification:

// Comparing unexported fields from packages that are not owned by the user
// is unsafe since changes in the internal implementation may cause the result
// of Equal to unexpectedly change. This option should only be used on types
// where the semantic meaning of unexported fields is in full control of the user.

Reply:
Done.